src/server/middleware/twitter.middleware.ts

```diff
@@ -304,6 +313,172 @@ function loadTwitterData(
+  }
+
+  const params = {
+    // Include next_token if it's part of the request.
+    ...(request.nextPageToken && { next_token: request.nextPageToken }),
+    ...{
+      query: `(@${user} OR url:twitter.com/${user}) -from:${user} -is:retweet`,
+      max_results: BATCH_SIZE,
+      'user.fields': 'id,name,username,profile_image_url,verified',
+      expansions: 'author_id,attachments.media_keys,referenced_tweets.id',
+      start_time: formatTimestampForV2(request.startDateTimeMs),
+      end_time: formatTimestampForV2(request.endDateTimeMs),
+      'media.fields': 'url,type',
+      'tweet.fields':
+        'attachments,created_at,id,entities,lang,public_metrics,source,text',
+    },
+  };
+
+  return axios
+    .get<TwitterApiResponse>(requestUrl, {
+      headers: {
+        authorization: `Bearer ${credentials.bearerToken}`,
+      },
+      params,
+      transformResponse: [
+        (data, _) => {
+          const parsed = JSON.parse(data);
+          const tweets: V2TweetObject[] = parsed.data ?? [];
+          const includes: V2Includes = parsed.includes ?? {};
+          return <TwitterApiResponse>{
+            results: tweets.map((tweet) => packV2Tweet(tweet, includes)),
+            next: parsed.meta.next_token,
+          };
+        },
+      ],
+    })
+    .then((response) => response.data);
+}
+
+// Packs a Tweet response object from the v2 Search API format into the
+// Enterprise Search API format.
+function packV2Tweet(tweet: V2TweetObject, includes: V2Includes): TweetObject {
+  const entities = packEntities(tweet, includes);
+
+  const tweetObject: TweetObject = {
+    created_at: tweet.created_at,
+    id_str: tweet.id,
+    text: tweet.text,
+    // The Enterprise API entities field is not always complete, but the v2
+    // entities field is.
+    entities: entities,
+    extended_entities: entities,
+    // `display_text_range` omitted because v2 does not truncate the text.
+    favorite_count: tweet.public_metrics.like_count,
+    // `favorited` omitted because it is not available in v2..
+    in_reply_to_status_id: tweet.entities?.referenced_tweets?.find(
+      (tweet) => tweet.type === 'replied_to'
+    )?.id,
+    lang: tweet.lang,
+    reply_count: tweet.public_metrics.reply_count,
+    retweet_count: tweet.public_metrics.retweet_count,
+    source: tweet.source,
+    truncated: tweet.text.length > 140,
+    user: getUser(tweet.author_id, includes),
+  };
+
+  if (tweetObject.truncated) {
+    // Enteprise populates the extended_tweet field if the tweet is truncated,
+    // so we add it manually here for consistency.
+    tweetObject.extended_tweet = {
+      full_text: tweetObject.text,
+      display_text_range: [0, 0], // Indices not available in v2.
+      entities,
+    };
+  }
+
+  return tweetObject;
+}
+
+function packEntities(
+  tweet: V2TweetObject,
+  includes: V2Includes
+): TweetEntities {
+  const entities: TweetEntities = {};
+
+  if (tweet.entities && tweet.entities.hashtags) {
```

Comment:
You used the ? syntax in a few other places; I think here would also be applicable.